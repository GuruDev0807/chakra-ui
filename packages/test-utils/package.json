{
  "name": "@chakra-ui/test-utils",
  "version": "1.0.0-next.6",
  "private": true,
  "description": "Testing utilities for Chakra UI's components",
  "keywords": [
    "chakra",
    "test utils",
    "testing",
    "utils",
    "testing library",
    "react testing library"
  ],
  "author": "Segun Adebayo <sage@adebayosegun.com>",
  "homepage": "https://github.com/chakra-ui/chakra-ui#readme",
  "license": "MIT",
  "main": "src/index.ts",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/chakra-ui/chakra-ui.git"
  },
  "bugs": {
    "url": "https://github.com/chakra-ui/chakra-ui/issues"
  },
  "dependencies": {
    "@chakra-ui/system": "^1.0.0-next.6",
    "@chakra-ui/theme": "^1.0.0-next.6",
    "@testing-library/jest-dom": "5.1.1",
    "@testing-library/react": "10.4.5",
    "@testing-library/react-hooks": "3.2.1",
    "@testing-library/user-event": "12.0.11",
    "@types/jest-axe": "3.2.1",
    "jest-axe": "3.4.0",
    "react-test-renderer": "16.12.0"
  },
  "peerDependencies": {
    "react": "16.x"
  }
}
