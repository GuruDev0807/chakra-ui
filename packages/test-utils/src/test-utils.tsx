import { ChakraProvider } from "@chakra-ui/core"
import "@testing-library/jest-dom/extend-expect"
import {
  render as RTLRender,
  RenderOptions,
  fireEvent,
  RenderResult,
  Queries,
} from "@testing-library/react"
import * as React from "react"
import { toHaveNoViolations } from "jest-axe"
import serializer from "jest-emotion"

expect.addSnapshotSerializer(serializer)
expect.extend(toHaveNoViolations)

const AllProviders: React.FC = ({ children }) => (
  <ChakraProvider resetCSS>{children}</ChakraProvider>
)

export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
): RenderResult

export function render(ui: React.ReactElement, options?: RenderOptions) {
  return RTLRender(ui, { wrapper: AllProviders, ...options })
}

export * from "@testing-library/react"

export {
  act as invoke,
  renderHook,
  RenderHookOptions,
  RenderHookResult,
} from "@testing-library/react-hooks"

export { default as userEvent } from "@testing-library/user-event"

export * from "jest-axe"

export const escape = (ui: HTMLElement) =>
  fireEvent.keyDown(ui, { key: "Escape", keyCode: 27 })
