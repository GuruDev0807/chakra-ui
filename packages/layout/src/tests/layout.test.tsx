import { render } from "@chakra-ui/test-utils"
import React from "react"
import { Box } from ".."

test("should render box correctly", () => {
  const tools = render(<Box>This is box</Box>)
  expect(tools.asFragment()).toMatchSnapshot()
})

test("as - prop works correctly", () => {
  const tools = render(
    <Box as="a" href="www.google.com">
      This is box
    </Box>,
  )
  expect(tools.asFragment()).toMatchSnapshot()
})
