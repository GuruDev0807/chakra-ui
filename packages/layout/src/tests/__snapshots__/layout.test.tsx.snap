// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as - prop works correctly 1`] = `
<DocumentFragment>
  <a
    class="css-0"
    href="www.google.com"
  >
    This is box
  </a>
</DocumentFragment>
`;

exports[`should render box correctly 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    This is box
  </div>
</DocumentFragment>
`;
